CesiumUtility/include/CesiumUtility/Uri.h

```diff
@@ -24,8 +24,105 @@ class Uri final {
       const std::string& templateUri,
       const std::function<SubstitutionCallbackSignature>& substitutionCallback);
 
+  /**
+   * @brief Escapes a portion of a URI, percent-encoding disallowed characters.
+   *
+   * @param s The string to escape.
+   * @return The escaped string.
+   */
   static std::string escape(const std::string& s);
 
+  /**
+   * @brief Unescapes a portion of a URI, decoding any percent-encoded
+   * characters.
+   *
+   * @param s The string to unescape.
+   * @return The unescaped string.
+   */
+  static std::string unescape(const std::string& s);
+
+  /**
+   * @brief Converts a Unix file system path to a string suitable for use as the
+   * path portion of a URI. Characters that are not allowed in the path portion
+   * of a URI are percent-encoded as necessary.
+   *
+   * If the path is absolute (it starts with a slash), then the URI will start
+   * with a slash as well.
+   *
+   * @param unixPath The file system path.
+   * @return The URI path.
+   */
+  static std::string unixPathToUriPath(const std::string& unixPath);
```

Comment:
I noticed these 3 functions are basically the same, other than the file system they target. (same for the next group of 3 that does the reverse).

Might be able to simplify the API a bit by just making the file system an argument...
```
enum class FileSystem {
  Unix,
  Windows,
  AutoDetect
};
static std::string fileSystemPathToUriPath(const std::string& path, FileSystem fileSystem = AutoDetect);
```

Reply:
Hmm it's an interesting idea, but it would add a tiny runtime cost and I'm not sure what we'd really be gaining.

Reply:
> not sure what we'd really be gaining

Not all that much. Just one function declared in the .h, rather than three (they could just live in the .cpp).